src/unix/fs.c

```diff
@@ -1253,22 +1277,38 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
       return UV__ERR(errno);
   }
 
-  flags = COPYFILE_ALL;
+  if ((req->flags & UV_FS_COPYFILE_FICLONE) ||
+      (req->flags & UV_FS_COPYFILE_FICLONE_FORCE)) {
+    rc = uv__fs_clonefile_mac(req);
 
-  if (req->flags & UV_FS_COPYFILE_FICLONE)
-    flags |= COPYFILE_CLONE;
+    /* Return on success.
+     * If an error occurred and force was set, return the error to the caller;
+     * fall back to copyfile() when force was not set. */
+    if (rc == 0 || req->flags & UV_FS_COPYFILE_FICLONE_FORCE)
```

Comment:
```suggestion
    if (rc == 0 || (req->flags & UV_FS_COPYFILE_FICLONE_FORCE))
```
Emits a compiler warning otherwise.